src/item_location.cpp

```diff
@@ -22,7 +22,7 @@ class item_location::impl
     public:
         virtual ~impl() = default;
         virtual std::string describe( const Character * ) const = 0;
-        virtual int obtain( Character & ) = 0;
+        virtual int obtain( Character &ch, int qty ) = 0;
```

Comment:
Asides from existing casts is there any reason why `item::charges` cannot become an `int`?